src/utils.js

```diff
@@ -51,7 +49,7 @@ function getFetchOptions(options) {
   delete headers.host;
   delete headers.connection;
   return {
-    cache: 'no-cache',
```

Comment:
This cache option was never supported by helix-fetch (v1 & v2). Using the correct `cache: 'no-store'` option causes the following test error: 

```
1) Index Tests
       index produces 404 when fetcher fails due to missing action (with 404 handler).:
     TypeError: Stream had error: The operation was aborted.
```

The same error was also triggered with helix-fetch v1 but was hidden by the incorrect cache option.

Seems like this code doesn't handle redirects correctly: 
https://github.com/adobe/helix-dispatch/blob/helix-v2/src/index.js#L227-L232

Reply:
no. this error was due to another problem: for the 404 page, we receive a 200 status, but need to tweak it to look like a 404. with helix-fetch v1, we could just override the `response.status` but with v2 this is protected (even worse, the object is not frozen, so no error is thrown, but even if you set res.status=404, it remains 200). so we wrap it with a `Proxy` to override the `status`. later in the `finally` block, it aborts all pending requests, if they are have not produced the response that is returned. since the _proxied_ response is now not the same response again that we check against, it was aborted.

Reply:
hmm, I debugged the test with helix-fetch v1 and `cache: 'no-store'`. I got the exact same failure (aborted). So it doesn't seem to be related to v2 vs v1.

Reply:
hmm... AFAICR, I added the `Proxy` recently, as I couldn't overwrite the status.... but I think it was for the universal deploy changes. this code is so complicated, there might be several layers of potential errors :-(

Reply:
> so we wrap it with a Proxy to override the status

Why not 

```
const { headers, url } = resp;
const resp404 = new Response(res.body, { status: 404, headers, url });
```

?

Reply:
I try...

Reply:
> but with v2 this is protected (even worse, the object is not frozen, so no error is thrown, but even if you set res.status=404, it remains 200)

FWIW: I agree, it is confusing. However it's the same behavior as with the most popular fetch nodejs module (`node-fetch`, 21m weekly downloads) and in the browser.